### ISSUE TITLE
Is there something like koffi.free() to release the dynamic allocated memory returned to Node.js

### ISSUE THREAD

**Issue by wanglin86769** [Koromix/koffi#126]
Hello,

The **data pointers** section of the koffi website mentions **disposable types** and `koffi.free(ptr)` function, in which **disposable types** is a little bit complicated for me and `koffi.free(ptr)` is exactly what I want, that is I just want sometime like `koffi.free(ptr)` to release the dynamic allocated memory (using malloc() in C++) returned to Node.js. However, `koffi.free(ptr)` does not seem to be a callable function since I cannot find it in index.d.ts.

For example, my use case is as follows,

C++ code:
```
typedef struct SimpleValue {
    int type;
    int count;
    void *buffer;
} SimpleValue;

void _getParam(const char* name, SimpleValue* simpleValue) {
    Value *value = driver->getParam(name);

    simpleValue->type = value->getType();
    simpleValue->count = value->getCount();
    simpleValue->buffer = value->getBuffer();  // Allocated using malloc()
}
```

Node.js code:
```
const SimpleValue = koffi.struct('SimpleValue', {
    type: 'int',
    count: 'int',
    buffer: 'void *'
});

const _getParam = libpcas.func('getParam', 'void', ['char *', koffi.out('SimpleValue *')]);

function getParam(name) {
    let simpleValue = {};
    _getParam(name, simpleValue);
    let array;
    switch(simpleValue.type) {
        case aitEnum.aitEnumInt32:
            array = koffi.decode(simpleValue.buffer, 'int', simpleValue.count);
            break;
        case aitEnum.aitEnumFloat32:
            array = koffi.decode(simpleValue.buffer, 'float', simpleValue.count);
            break;
        case aitEnum.aitEnumFloat64:
            array = koffi.decode(simpleValue.buffer, 'double', simpleValue.count);
            break;
        case aitEnum.aitEnumString:
            array = bufferToStringArray(simpleValue.buffer, simpleValue.count);
            break;
        case aitEnum.aitEnumEnum16:
            array = koffi.decode(simpleValue.buffer, 'int', simpleValue.count);
            break;
        default:
            console.log(`getParam(): Unknown PV type ${simpleValue.type} for PV ${name}`);
            return null;
    }


    koffi.free(simpleValue.buffer);  // I need something like koffi.free() here


    return simpleValue.count === 1 ? array[0] : array;
}
```

What is the recommended way to achieve this purpose?

**Comment by Koromix** [comment]
Just forgot to put `koffi.free()` in the TypeScript definition file :) I don't use TypeScript myself, so it's best effort at this point.

Added it here:
https://github.com/Koromix/rygel/commit/827c0728cb65805d07a32b092b8d5e6e2cfcdd32

Just hack it in your `node_modules/koffi/index.d.ts` until next release!

**Comment by Koromix** [comment]
The fix is in Koffi 2.7.2!
